script.js

```diff
@@ -38,7 +38,7 @@ $(function () {
 
     var onChange = function (event) {
         var checkbox = event.currentTarget;
-        if ($(checkbox).prev().hasClass("lock")) {
+        if ($(checkbox).next().hasClass("lock") || $(checkbox).next().hasClass("last-number")) {
```

Comment:
Looks like this is what's preventing the row from being lockable when it should be. That said, the old line certainly wasn't clear in what it was checking, so definitely warrants some investigating.